apps/full-site-editing/full-site-editing-plugin/full-site-editing/store/selectors.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * External dependencies
+ */
+import { createRegistrySelector } from '@wordpress/data';
+import { get } from 'lodash';
+
+export const isFullSitePage = createRegistrySelector( select => () => {
+	const { getBlocks, getCurrentPostType, getEditedPostAttribute } = select( 'core/editor' );
+
+	const postType = getCurrentPostType();
+	const templateId = get( getEditedPostAttribute( 'meta' ), '_wp_template_id' );
```

Comment:
I little bit late to the party but - this should always work for page post type, since every page will have a global page template implicitly assigned when the plugin is activated. In this case that would mean that `hasTemplate` is always true for pages, provided that the plugin is active.